apps/ledger-live-mobile/src/newArch/features/Web3Hub/components/ManifestsList/ManifestItem/index.tsx

```diff
@@ -83,35 +102,26 @@ export default function ManifestItem({
       <Flex flexDirection="row" alignItems="center" height={72} paddingX={4} paddingY={2}>
         <AppIcon isDisabled={isDisabled} size={48} name={manifest.name} icon={icon} />
         <Flex marginX={16} height="100%" flexGrow={1} flexShrink={1} justifyContent={"center"}>
-          <Flex flexDirection="row" alignItems={"center"} mb={2}>
+          <Flex flexDirection="row" alignItems={"center"}  mb={2}  columnGap={4}>
             <Text variant="large" color={color} numberOfLines={1} fontWeight="semiBold">
               {manifest.name}
             </Text>
-            {manifest.branch !== "stable" && (
-              <Text
-                fontSize="9px"
-                width="auto"
-                paddingX={2}
-                paddingY={1}
-                borderWidth={1}
-                borderRadius={3}
-                borderStyle={"solid"}
-                flexGrow={0}
-                flexShrink={0}
-                marginLeft={3}
-                overflow={"hidden"}
-                textTransform="uppercase"
-                color={badgeColor}
-                borderColor={borderColor}
-                backgroundColor={backgroundColor}
-                fontWeight="semiBold"
-              >
-                {t(`platform.catalog.branch.${manifest.branch}`, {
+            <Flex flexDirection="row" alignItems={"center"} columnGap={4}>
+            {manifest.branch !== "stable" &&
+              label(
+                t(`platform.catalog.branch.${manifest.branch}`, {
                   defaultValue: manifest.branch,
-                })}
-              </Text>
-            )}
+                }),
+                getItemStyle(manifest.branch, colors)
```

Comment:
Why would you call that again while we already call that in a memo above ?
I'd prefer to not confuse and merge a style based on the branch with a base style not based on it at all for another not related badge